addon/index.js

```diff
@@ -77,6 +75,51 @@ function updatePrefs() {
   });
 }
 
+// Listen for metrics events from experiments.
+const metrics = {
+  isInitialized: false,
+  init: function() {
+    if (this.isInitialized) {
+      return;
+    }
+    this.isInitialized = true;
+    // Note: the observer service holds a strong reference to this observer,
+    // so we must detach it on shutdown / uninstall by calling destroy().
+    Services.obs.addObserver(metrics, 'idea-town::send-metric', false);
```

Comment:
Nit: Might be handy to stick `idea-town::send-metric` in a CONSTANT var near the start of the file